src/printer.js

```diff
@@ -64,18 +67,18 @@ const printTopLevelParts = (node, path, opts, print) => {
   }
 
   const docs = flatten([parts.frontmatter, ...parseSortOrder(opts.astroSortOrder).map((p) => parts[p])]).filter((doc) => '' !== doc);
-  return group([join(hardline, docs)]);
```

Comment:
Removing the `hardline` cut down on too much whitespace in areas (see prev. comment).

Reply:
See https://github.com/snowpackjs/prettier-plugin-astro/pull/31